SOLIDITY_STYLE_GUIDE.md

```diff
@@ -1851,3 +1853,57 @@ function processStaking(
     ...
 }
 ```
+## Security
+
+### Checks Effects Interactions
+
+Checks effects interactions pattern should be used for writing contract methods which make external calls. This pattern reduces the attack surface for malicious contracts trying to hijack control flow after an external call.
+
+When to use this pattern: 
+
+* It cannot be avoided to hand over control flow to an external entity.
+* You want to protect your functions against re-entrancy attacks.
+
+Details:
+
+1. **Checks**: If input is not acceptable then fail fast and fail hard. 
+
+    ```
+    require(someCondition);
+    
+    ```
+
+2. **Effects**: Optimistically update the contract to a new valid state assuming that interactions will be successful. This protects the contract from re-entrance and race conditions.
+
+    ```
+    balance[msg.sender] = 0;
+    
+    ```
+
+3. **Interactions**: External function calls, `.send()`, `.transfer()` and `.call()` should be done at the end of the method. The return value of external call should be asserted if possible. 
+
+    ```
+    require(otherContract.doSomething());
+
+    ```
+
+
+Example: 
+```
+function withdraw(uint256 amount) public {
+    require(balances[msg.sender] >= amount); // Checks
+
+    balances[msg.sender] -= amount;          // Effects
+
+    require(                                 // Interactions 
+        msg.sender.send(amount),
+        "Fund transfer failed."
+    );
+}
+```
+
+**Note**: Even if emitting an event is technically considered as state change, it should be done at the end of the method in favor of readability. Position of event generally doesn't impact security aspect of the contract. 
+
+
+Refer [this](https://github.com/fravoll/solidity-patterns/blob/master/docs/checks_effects_interactions.md) for more details.
```

Comment:
My preference would be for "Checks effects interactions" at the start of L1860 to be hyperlinked to the reference source and for this line to be struck.

But I will defer to your assessment of reader-friendliness, because I don't think we have yet established a practice.

Reply:
I don't have a strong opinion here.  I thought giving references to the end so that the reader doesn't go down the rabbit hole in the beginning.